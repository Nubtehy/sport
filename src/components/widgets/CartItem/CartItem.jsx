import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';
import { CartItemRow, CartItemCol, QuantityContainer } from './CartItem.styled';

const displayName = 'CartItem';

const propTypes = {
  id: PropTypes.number.isRequired,
};

function CartItem({
  name,
  image,
  quantity,
  id,
  price,
  handlePlusItem,
  handleMinusItem,
  controls,
}) {
  return (
    id && (
      <CartItemRow data-name={id} key={id}>

        <CartItemCol><img src={image} alt="" /></CartItemCol>
        <CartItemCol>{name}</CartItemCol>
        <CartItemCol>{price}</CartItemCol>
        <CartItemCol>
          {
            controls ? (
              <div>
                <Button onClick={handleMinusItem}>-</Button>
                <QuantityContainer>{quantity}</QuantityContainer>
                <Button onClick={handlePlusItem}>+</Button>
              </div>
            ) : quantity
          }
        </CartItemCol>
        <CartItemCol>
          {price * quantity}
        </CartItemCol>
      </CartItemRow>
    )
  );
}

CartItem.displayName = displayName;
CartItem.propTypes = propTypes;

export default CartItem;
