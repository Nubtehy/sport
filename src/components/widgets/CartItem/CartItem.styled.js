import styled from 'styled-components';

export const CartItemRow = styled.tr`
  font-size: 14px;
`;
export const CartItemCol = styled.td`
  width: 20%;
  text-align: center;
  padding: 10px;
  border-bottom: 1px solid #cccccc;
`;

export const QuantityContainer = styled.span`
  font-size: 18px;
  font-weight: bold;
`;
