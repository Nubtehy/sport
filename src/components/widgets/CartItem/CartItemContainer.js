import {
  compose, setDisplayName, withHandlers, setPropTypes,
} from 'recompose';
import PropTypes from 'prop-types';
import CartItem from './CartItem';


export const enhance = compose(
  setDisplayName('ProductContainer'),
  setPropTypes({
    id: PropTypes.number.isRequired,
    minus: PropTypes.func,
    plus: PropTypes.func,
    control: PropTypes.bool,
  }),
  withHandlers({
    handleMinusItem: ({ id, minus }) => () => minus(id),
    handlePlusItem: ({ id, plus }) => () => plus(id),
  }),
);

export default enhance(CartItem);
