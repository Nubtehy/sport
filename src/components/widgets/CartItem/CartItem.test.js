import React from 'react';
import { shallow } from 'enzyme';
import CartItem from './CartItem';

const testProps = {
  name: 'Product 1',
  image: '',
  quantity: 1,
  id: 1,
  price: 22,
  handlePlusItem: jest.fn(),
  handleMinusItem: jest.fn(),
  controls: true,
};
describe('Given the CartItem component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<CartItem {...testProps} />);
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
