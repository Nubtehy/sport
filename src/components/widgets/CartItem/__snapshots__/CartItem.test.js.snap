// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the CartItem component when the component is rendered should render the component with provided props 1`] = `
<CartItemstyled__CartItemRow
  data-name={1}
  key="1"
>
  <CartItemstyled__CartItemCol>
    <img
      alt=""
      src=""
    />
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    Product 1
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    22
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    <div>
      <WithStyle(Button)
        onClick={[MockFunction]}
      >
        -
      </WithStyle(Button)>
      <CartItemstyled__QuantityContainer>
        1
      </CartItemstyled__QuantityContainer>
      <WithStyle(Button)
        onClick={[MockFunction]}
      >
        +
      </WithStyle(Button)>
    </div>
  </CartItemstyled__CartItemCol>
  <CartItemstyled__CartItemCol>
    22
  </CartItemstyled__CartItemCol>
</CartItemstyled__CartItemRow>
`;
